libraries/core_libs/network/src/tarcap/packets_handlers/dag_block_packet_handler.cpp

```diff
@@ -104,7 +104,12 @@ void DagBlockPacketHandler::onNewBlockReceived(DagBlock &&block, const std::shar
         throw MaliciousPeerException(err_msg.str());
       }
       case DagManager::VerifyBlockReturnType::MissingTransaction:
-        if (peer->peer_dag_synced_) {
+        if (trx_mgr_->isTransactionPoolFull()) [[unlikely]] {
```

Comment:
I think in some edge cases we might not be full but still get into this state because we were recently full. Maybe we should do this even if we are 80% or 90% full?

Reply:
good point, let's set it to 90%